From 48c66a5e8e0725c459cc7a0b5a2bb4661d47d025 Mon Sep 17 00:00:00 2001
From: wfeng6 <wei.feng@intel.com>
Date: Fri, 3 Aug 2012 00:44:30 +0800
Subject: [PORT FROM MAIN] To fix the green bar problem in video editor

BZ: 34031

This is an Android code defect. Should use video stride to
clear the video data buffer instead of the video width.

Change-Id: Iee3e39d2ba342d9a318add8c2c22510c60f4b950
Orig-Change-Id: I341eb4db5f694606aa833c5969b095eb7510e86a
Signed-off-by: wfeng6 <wei.feng@intel.com>
Signed-off-by: Dan Liang <dan.liang@intel.com>
---
 .../lvpp/VideoEditorPreviewController.cpp          |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/libvideoeditor/lvpp/VideoEditorPreviewController.cpp b/libvideoeditor/lvpp/VideoEditorPreviewController.cpp
index 149c4ea..76f1700 100755
--- a/libvideoeditor/lvpp/VideoEditorPreviewController.cpp
+++ b/libvideoeditor/lvpp/VideoEditorPreviewController.cpp
@@ -779,11 +779,11 @@ M4OSA_ERR VideoEditorPreviewController::clearSurface(
      (M4OSA_UInt32)outBufferStride, (M4VIFI_UInt8 *)outBuffer);
 
     /* Fill the surface with black frame */
-    memset((void *)planeOut[0].pac_data,0x00,planeOut[0].u_width *
+    memset((void *)planeOut[0].pac_data,0x00,planeOut[0].u_stride *
                             planeOut[0].u_height * 1.5);
-    memset((void *)planeOut[1].pac_data,128,planeOut[1].u_width *
+    memset((void *)planeOut[1].pac_data,128,planeOut[1].u_stride *
                             planeOut[1].u_height);
-    memset((void *)planeOut[2].pac_data,128,planeOut[2].u_width *
+    memset((void *)planeOut[2].pac_data,128,planeOut[2].u_stride *
                              planeOut[2].u_height);
 
     mTarget->renderYV12();
-- 
1.7.4.1

